demos/2019_RHTE/labs/8.md

```diff
@@ -0,0 +1,254 @@
+# 8.0 Migration at Scale via API (optional)
+
+In this section, we will be looking at migrating multiple namespaces in a single Migration Plan.  In these types of large scale migrations, sometimes using the WebUI can be less than ideal.  With that, we are also going to use the API directly for creating and executing our migration.
+
+## 8.1 Setup
+
+Let's start by seeding our 3.11 cluster with a set of sample stateless applications.  We've added a set of [scripts](./scripts/lab8/) to make this simple:
+
+### 8.1.0 Deploy HelloOpenShift Apps
+
+The `deploy.sh` script will deploy the HelloOpenShift application to a user specified number of namespaces.  For our purposes, let's choose 5.
+
+```bash
+$ ./deploy.sh
+Number of namespaces?
+5
+namespace/hello-openshift-1 created
+pod/hello-openshift created
+service/hello-openshift exposed
+route.route.openshift.io/hello-openshift exposed
+namespace/hello-openshift-2 created
+pod/hello-openshift created
+service/hello-openshift exposed
+route.route.openshift.io/hello-openshift exposed
+namespace/hello-openshift-3 created
+pod/hello-openshift created
+service/hello-openshift exposed
+route.route.openshift.io/hello-openshift exposed
+namespace/hello-openshift-4 created
+pod/hello-openshift created
+service/hello-openshift exposed
+route.route.openshift.io/hello-openshift exposed
+namespace/hello-openshift-5 created
+pod/hello-openshift created
+service/hello-openshift exposed
+route.route.openshift.io/hello-openshift exposed
+Finding routes...
+hello-openshift-hello-openshift-1.apps.db45.events.opentlc.com
+hello-openshift-hello-openshift-2.apps.db45.events.opentlc.com
+hello-openshift-hello-openshift-3.apps.db45.events.opentlc.com
+hello-openshift-hello-openshift-4.apps.db45.events.opentlc.com
+hello-openshift-hello-openshift-5.apps.db45.events.opentlc.com
+```
+
+### 8.1.1 Validate applications
+
+We've also included a `probe.sh` script that will verify that all deployed applications are responding.
+
+```bash
+$ ./probe.sh
+Probing app in namespace hello-openshift-1
+Hello OpenShift!
+Probing app in namespace hello-openshift-2
+Hello OpenShift!
+Probing app in namespace hello-openshift-3
+Hello OpenShift!
+Probing app in namespace hello-openshift-4
+Hello OpenShift!
+Probing app in namespace hello-openshift-5
+Hello OpenShift!
+```
+
+Great!  We are now ready to assemble our Migration Plan.
+
+## 8.2 Create Migration Plan
+
+Since we've already performed successful migrations throughout the preceding labs, we already have our MigCluster, Cluster, & MigStorage resources created. This means we are ready to assemble our MigPlan.
```

Comment:
Just a note that 'Cluster' is likely disappearing soon, might exist for labs, then pulled out for GA.